operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -291,10 +308,35 @@ func (r *TenantReconciler) checkValidWorkspaces(ctx context.Context, tn *crownla
 			tnOpinternalErrors.WithLabelValues("tenant", "workspace-not-exist").Inc()
 		} else {
 			tenantExistingWorkspaces = append(tenantExistingWorkspaces, tnWs)
-			workspaces = append(workspaces, ws)
+			if tnWs.Role == crownlabsv1alpha2.Candidate {
+				if !slices.Contains(possibleEnrollFeatures, string(ws.Spec.AutoEnroll)) {
+					// autoEnroll is not enabled, so automatically remove the workspace from the tenant
+					klog.Infof("Workspace %s is not autoEnroll enabled, removing it from tenant %s", tnWs.Name, tn.Name)
+					tn.Spec.Workspaces = r.removeWorkspace(tn.Spec.Workspaces, tnWs.Name)
```

Comment:
Here you're editing something you're ranging onto, iirc it's quite dangerous, I'd suggest to make an array of workspaces that need to be updated, then, after this, you range onto this new array twice: first time to remove them from the `tn.Spec`, then to add them back...

Reply:
I've inserted two arrays (one for workspaces I have to remove and another for ones I have to add)